TaskRun details and log page

abhinandan13jan opened the pull request:
# Addresses
https://issues.redhat.com/browse/ODC-4801

# Screenshot
![task-run-det-multi](https://user-images.githubusercontent.com/24852534/96107095-9c7cc400-0ef9-11eb-951b-63a2982153b7.gif)

# Tests
Added two test suites, TaskRunDetails and TaskRunLogs

# Browser conformance
chrome

karthikjeeyar commented on frontend/packages/dev-console/src/components/taskruns/TaskRunLog.tsx:
```diff
@@ -0,0 +1,30 @@
+
+// to safeguard against cases where Pod is yet to be assigned or is removed
+export default (props: TaskRunLogProps) =>
+  props.obj.status?.podName ? (
+    <TaskRunLog {...props} />
+  ) : (
+    <StatusBox loadError="Pod not found" label="TaskRun Log" />
```
Why can't we have this check in the above function and early exit if podName is not available :thinking:

abhinandan13jan replied:
It works, but it fails lint check when there is an `if` block before the `useK8sGet` Hook

sahil143 replied:
Best to name this wrapper component and export that as default.

abhinandan13jan replied:
named as wrapper

karthikjeeyar commented:
/kind feature

sahil143 commented on frontend/packages/dev-console/src/components/taskruns/TaskRunLog.tsx:
```diff
@@ -0,0 +1,30 @@
+    obj.status.podName,
+    obj.metadata.namespace,
+  );
+  if (podLoaded) {
```
```suggestion
  if (podLoaded && !podError) {
```

abhinandan13jan replied:
updated this as well as named to `TaskRunLogWrapper`

sahil143 commented on frontend/packages/dev-console/src/components/taskruns/TaskRunLog.tsx:
```diff
@@ -0,0 +1,32 @@
+};
+
+// to safeguard against cases where Pod is yet to be assigned or is removed
+const TaskRunLogWrapper = (props: TaskRunLogProps) =>
```
This is a Functional Component. Should type it properly
```suggestion
const TaskRunLogWrapper: React.FC<TaskRunLogProps> = (props) =>
```

sahil143 reviewed (commented):
@abhinandan13jan Logs doesn't seem to be streaming. It just says loading and when step is changed from the dropdown. There is an error
<img width="1485" alt="Screenshot 2020-10-09 at 10 23 47 AM" src="https://user-images.githubusercontent.com/9278015/95545157-b5c5d200-0a1a-11eb-8ea0-07df83269136.png">
<img width="1482" alt="Screenshot 2020-10-09 at 10 24 00 AM" src="https://user-images.githubusercontent.com/9278015/95545163-b9595900-0a1a-11eb-80b9-656d94e02a8c.png">
But it streams fine in the Pipeline run logs
<img width="1490" alt="Screenshot 2020-10-09 at 10 24 13 AM" src="https://user-images.githubusercontent.com/9278015/95545218-e60d7080-0a1a-11eb-8f97-d72d476e0f3d.png">

karthikjeeyar commented on frontend/packages/dev-console/src/components/taskruns/TaskRunDetails.tsx:
```diff
@@ -15,6 +19,15 @@ const TaskRunDetails: React.FC<TaskRunDetailsProps> = ({ obj: taskRun }) => {
+          <dl>
+            <dt>Status</dt>
+            <dd>
+              <Status status={pipelineRunFilterReducer(taskRun)} />
+            </dd>
+          </dl>
+          <PipelineRunDetailsErrorLog pipelineRun={taskRun} />
```
These two pipelineRun specific functions called in the context of `taskRun` looks odd to me. If taskRun has more specific type definition, then this will cause lint errors and confusions in future.  It will be good to refactor these to a generic function calls. WDYT?

abhinandan13jan replied:
I've updated such that PLR and TR uses separate utils and point to a common component

abhinandan13jan commented:
> @abhinandan13jan Logs doesn't seem to be streaming. It just says loading and when step is changed from the dropdown. There is an error
> <img alt="Screenshot 2020-10-09 at 10 23 47 AM" width="1485" src="https://user-images.githubusercontent.com/9278015/95545157-b5c5d200-0a1a-11eb-8ea0-07df83269136.png">
> <img alt="Screenshot 2020-10-09 at 10 24 00 AM" width="1482" src="https://user-images.githubusercontent.com/9278015/95545163-b9595900-0a1a-11eb-80b9-656d94e02a8c.png">
> But it streams fine in the Pipeline run logs
> <img alt="Screenshot 2020-10-09 at 10 24 13 AM" width="1490" src="https://user-images.githubusercontent.com/9278015/95545218-e60d7080-0a1a-11eb-8f97-d72d476e0f3d.png">

Strange, I find it to work the same as Pod log

![trlog2](https://user-images.githubusercontent.com/24852534/95559595-a3a55d00-0a35-11eb-89bf-c6c082005a97.gif)
![trlog1](https://user-images.githubusercontent.com/24852534/95559608-aacc6b00-0a35-11eb-8fcb-ccd868817d57.gif)

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/logs/LogSnippetBlock.tsx:
```diff
@@ -1,19 +1,30 @@
 type LogSnippetBlockProps = {
   children: (logSnippet: string) => React.ReactNode;
   logDetails: PipelineRunErrorDetails;
-  pipelineRun: PipelineRun;
+  pipelineRun?: PipelineRun;
+  taskRun?: K8sResourceKind;
```
Maybe we just simplify this to a single `K8sResourceKind` or even better down to just `namespace`?

abhinandan13jan replied:
simplified to namespace for all cases

andrewballantyne commented on frontend/packages/dev-console/src/components/taskruns/__tests__/TaskRunDetails.spec.tsx:
```diff
@@ -0,0 +1,42 @@
+  },
+};
+
+describe('TaskRunDetailsPage', () => {
+  const pipelineRunWrapper = shallow(<TaskRunDetails {...TaskDetailsProps} />);
+  it('Renders a SectionHeading', () => {
+    expect(pipelineRunWrapper.find(SectionHeading).exists());
+  });
+  it('Renders a Resource Summary', () => {
+    expect(pipelineRunWrapper.find(ResourceSummary).exists());
+  });
+  it('Renders a Status', () => {
+    expect(pipelineRunWrapper.find(Status).exists());
+  });
+  it('Renders a Task Run Log snippet', () => {
+    expect(pipelineRunWrapper.find(TaskRunDetailsErrorLog).exists());
+  });
+});
```
Testing against static content won't save us from anything. The best it can do is fail a test when we change the layout... there isn't any conditional here that we would be concerned about. Ie, no logic breaks can cause these tests to fail.

abhinandan13jan replied:
This is just based on current acceptance criteria. Just to make sure, the code adheres to the AC

andrewballantyne replied:
Unit tests guard against accidental breaks in logic.

Eg. A unit test for `add5()` would be `add5(5).toBe(10)`... you could test positive numbers, negative numbers, strings (should return NaN), objects, etc, etc...

If you wrap your `add5()`:

```
function getOffsetNumber() {
  return add5(37);
}
```
This function doesn't get you much but it allows the caller not to worry about providing this static `37` (this is an overly simple example). A testing of this function doesn't mean much. `getOffsetNumber().toBe(42)`... how will this test break? If we choose `37` is not a valid constant anymore? So we just change the test and the code at the same time and nothing super shocking here happens. No assistance is really provided by our test, no assurances we don't accidentally break.

Tests are so you can go into `add5` and change the fact that it doesn't add 5 anymore, it subtracts 8 or does something else (uses math.random for instance). Zero inputs or inputs that don't change the output are not test worthy scenarios. They just explicitly tie our tests to our implementation and thus makes us update one more place when we need to change something. It doesn't guard against breaks.

With this mental model, React Components have a similar structure with 1 extra condition. We only shallowly look at the structure of a component. So if your React Component does not take any props or the props are not conditional rendering props but just pass-thru props, then there are no tests needing to be written. 

So in summary, because I ranted a bit here. Don't test a component that does not use conditional rendering as you're not really writing a test, you're writing just 1 more spot to update if we need to change anything.

andrewballantyne commented on frontend/packages/dev-console/src/components/taskruns/TaskRunLog.tsx:
```diff
@@ -0,0 +1,32 @@
+import { PodModel } from '@console/internal/models';
+
+export type TaskRunLogProps = {
+  obj: K8sResourceKind;
```
```suggestion
  obj: TaskRunKind;
```

abhinandan13jan replied:
updated

karthikjeeyar reviewed (commented):
I observe the same issue as sahil mentioned above.When you create a taskRun, logs are not streaming correctly.

abhinandan13jan commented on frontend/packages/dev-console/src/components/taskruns/__tests__/TaskRunLog.spec.tsx:
```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { PodLogs } from '@console/internal/components/pod-logs';
+import { StatusBox } from '@console/internal/components/utils/status-box';
+import { shallow } from 'enzyme';
```
updated

abhinandan13jan commented on frontend/packages/dev-console/src/components/taskruns/__tests__/TaskRunDetails.spec.tsx:
```diff
@@ -0,0 +1,42 @@
+    },
+  },
+};
+
```
moved

abhinandan13jan commented on frontend/packages/dev-console/src/utils/pipeline-filter-reducer.ts:
```diff
@@ -58,3 +58,8 @@ export const pipelineResourceTypeFilter = (filters, pipelineResource): boolean =
 };
+
+export const taskRunFilterReducer = (taskRun): string => {
+  const status = pipelineRunStatus(taskRun);
```
There's one issue with that. It's a cyclic dependence and pipeline-filter-reducer doesn't have types. It should be a separate issue to refactor `Types` out of pipeline-augment. cc: @andrewballantyne

jerolimov replied:
I would prefer here to not reuse `pipelineRunStatus` at all and to not mix PipelineRun and TaskRun objects. I personally would prefer in this case "duplicated" code for both types instead of methods which supports both.

andrewballantyne replied:
Yeah, types are annoying in this file. So it should be handled by the work to take Pipelines out of DevConsole package

abhinandan13jan commented on frontend/packages/dev-console/src/utils/pipeline-filter-reducer.ts:
```diff
@@ -58,3 +58,8 @@ export const pipelineResourceTypeFilter = (filters, pipelineResource): boolean =
   return filters.selected.has(type) || !_.includes(filters.all, type);
 };
+
+export const taskRunFilterReducer = (taskRun): string => {
```
cyclic dependence will be there

abhinandan13jan commented:
> I observe the same issue as sahil mentioned above.When you create a taskRun, logs are not streaming correctly.

Fixed it with useK8sWatchResource hook

bgliwa01 commented:
Will all of the columns be added on the list view? Right now I only see namespace and created, but there should be others included as seen in this PR - https://github.com/openshift/console/pull/6840#event-3844836192 

I'm not sure if these designs will automatically update based on Deb's PR or not...

serenamarie125 reviewed (changes requested):
Based on @siamaksade comment ... 
> Strange, I find it to work the same as Pod log

@abhinandan13jan If possible, I'd suggest that we use the same component from the Log tab in PipelineRun here, but just not show the left nav piece with the task list

I think this was a miscommunication because we discussed "using conventions" :)

abhinandan13jan commented:
> Will all of the columns be added on the list view? Right now I only see namespace and created, but there should be others included as seen in this PR - [#6840 (comment)](https://github.com/openshift/console/pull/6840#event-3844836192)
> 
> I'm not sure if these designs will automatically update based on Deb's PR or not...

@bgliwa01 Yes, the list view will be updated as per @debsmita1 's PR. You can ignore the list view shown in my screenshots :)

abhinandan13jan commented:
> Based on @siamaksade comment ...
> 
> > Strange, I find it to work the same as Pod log
> 
> @abhinandan13jan If possible, I'd suggest that we use the same component from the Log tab in PipelineRun here, but just not show the left nav piece with the task list
> 
> I think this was a miscommunication because we discussed "using conventions" :)

I've updated to use the Multistreaming Log

![task-run-det-multi](https://user-images.githubusercontent.com/24852534/96107221-ca620880-0ef9-11eb-8dd2-c1709a480ac0.gif)

karthikjeeyar commented:
Verified locally, works fine
/lgtm

bgliwa01 reviewed (approved):
lgtm

invincibleJai commented:
@abhinandan13jan looks like, it needs rebase

jerolimov commented on frontend/packages/dev-console/src/components/taskruns/TaskRunDetailsErrorLog.tsx:
```diff
@@ -0,0 +1,35 @@
+};
+
+const TaskRunDetailsErrorLog: React.FC<TaskStatusLogProps> = ({ taskRun }) => {
+  const logDetails = getLogSnippet(taskRun);
```
`getLogSnippet` only accepts `PipelineRun` instances and no `TaskRun` types here. The getLogSnippet method converts the given PipelineRun to a logDetails object which will be used by `LogSnippetBlock` to fetch the correct log.

For a PipelineRun this method return an object like this one:

```tsx
{
  containerName: "step-build",
  podName: "docker-bs1qu0-build-qmsfx-pod-5hpp4",
  title: "Failure on task build - check logs for details.",
}
```

But for the TaskRun it returns this message (extracted from yaml):

```tsx
{
  staticMessage: "\"step-build\" exited with code 1 (image: \"registry.redhat.io/rhel8/buildah@sha256:180c4d9849b6ab0e5465d30d4f3a77765cf0d852ca1cb1efb59d6e8c9f90d467\"); for logs run: kubectl -n christoph-test logs docker-bs1qu0-build-qmsfx-pod-5hpp4 -c step-build\n",
  title: "Failure - check logs for details.",
}
```

This is the reason why we see currently a different error log does on the PipelineRun Detail page and on the TaskRun detail page.

The PipelineRun detail page fetches the log from the Pod:

![image](https://user-images.githubusercontent.com/139310/96659023-45864d00-1346-11eb-9e2c-21ef1edaeb43.png)

But the TaskRun detail page shows an error message from the yaml:

![image](https://user-images.githubusercontent.com/139310/96659091-664ea280-1346-11eb-9726-46be726cdb2c.png)

I think we should write a similar getLogSnippet (or getLogDetails) for TaskRuns which also extracts the right podName and containerName (and shows the current message also as fallback).

The status steps contains all required infos for this:

```yaml
status:
  completionTime: '2020-10-21T00:24:58Z'
  conditions:
    - message: >
        "step-build" exited with code 1.... # FALLBACK
      reason: Failed
      status: 'False'
      type: Succeeded
  podName: docker-bs1qu0-build-qmsfx-pod-5hpp4
  steps:
    - container: step-generate
      name: generate
      terminated:
        exitCode: 0
        reason: Completed
    - container: step-build
      name: build
      terminated:
        exitCode: 1
        reason: Error    # I expect that we want see the log snippet of this container
```

jerolimov replied:
Could verify that this is solved now :+1:

jerolimov reviewed (changes requested):
/hold

abhinandan13jan commented:
![Screenshot from 2020-10-22 15-26-08](https://user-images.githubusercontent.com/24852534/96856831-fc381980-147b-11eb-878a-31b11eaa8f45.png)

@jerolimov Thanks for pointing it out. I've separated out and refactored to use a new logsnippet for TaskRuns
@invincibleJai Rebased
@karthikjeeyar could you please re-review once the tests are done.

jerolimov commented:
Tested this locally and it works fine for me. There are an option question from Karthik about an a missing Type. Andrew was cc-ed and I added my opinion as well, otherwise I would give a lgtm here.

/hold cancel

jerolimov commented:
/lgtm

Let us see what @andrewballantyne says to the type definitions in pipeline-filter-reducer.ts

andrewballantyne commented on frontend/packages/dev-console/src/components/taskruns/events/__tests__/TaskRunDetailsPage.spec.tsx:
```diff
@@ -36,7 +36,7 @@ describe('TaskRunDetailsPage:', () => {
     const wrapper = shallow(<TaskRunDetailsPage {...taskRunDetailsPageProps} />);
     const { pages } = wrapper.props();
-    expect(pages).toHaveLength(3);
+    expect(pages).toHaveLength(4);
```
Point in case to a bad test. Did we worry about the number of pages dynamically changing on us? You just added a new tab, and then went an updated a test, likely after it failed on you and not as a proactive measure to drive a TDD scenario... so, what value does this test get us?

abhinandan13jan replied:
This was added as part of some separate PR and I felt simply to update it. I've removed the block. @andrewballantyne

andrewballantyne replied:
I understand this was written in another PR, but it was an example to help you understand what happens when you write a "static verification" test. There isn't really any gain here, nothing is protected from accidental breaks, and you just end up modifying the test as you change the content.

andrewballantyne reviewed (changes requested):
I'm seeing a lot of DRY problems here.

1. `getLogSnippet` for PipelineRun and TaskRun have similar logic, we _might_ be able to improve this a bit
2. `joinConditions` is repeated in both `taskRunLogSnippet.ts` and `pipelineRUnLogSnippet.ts`
3. The types are duplicated between both log file... we should have a generic type -- this also leads to `LogSnippetBlock` having the type `PipelineRunErrorDetails` for `logDetails` of which you pass a TaskRunErrorDetails value to

There are also issues with the tests...

- We have static tests tied making sure static content is rendered
- We have tests looking into the contents of another component to see if they are getting the right values
- We have `pipelineRunWrapper` in TaskRun tests

abhinandan13jan commented:
> I'm seeing a lot of DRY problems here.
> 
> 1. `getLogSnippet` for PipelineRun and TaskRun have similar logic, we _might_ be able to improve this a bit
> 2. `joinConditions` is repeated in both `taskRunLogSnippet.ts` and `pipelineRUnLogSnippet.ts`
> 3. The types are duplicated between both log file... we should have a generic type -- this also leads to `LogSnippetBlock` having the type `PipelineRunErrorDetails` for `logDetails` of which you pass a TaskRunErrorDetails value to
> 
> There are also issues with the tests...
> 
> * We have static tests tied making sure static content is rendered
> * We have tests looking into the contents of another component to see if they are getting the right values
> * We have `pipelineRunWrapper` in TaskRun tests

@andrewballantyne I've refactored and moved common `utils` from the `logSnippets` into a separate file and the common `types` as well. 
I've removed the `static tests from the DetailsPage component` and have added tests for the logSnippet utils.
I've also removed the number of tabs tests which was part of another PR.

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/logs/pipelineRunLogSnippet.ts:
```diff
@@ -4,26 +4,11 @@ import {
 } from '../../../utils/pipeline-augment';
+import { CombinedErrorDetails } from './log-snippet-types';
 import { pipelineRunStatus } from '../../../utils/pipeline-filter-reducer';
+import { taskRunSnippetMessage } from './log-snippet-utils';
```
You created a generic `log-snippet-utils`... perhaps we just move this content to that file instead of having a PLR one and a "generic" one that only contains TaskRun functionality. Just merge the two and we don't need to worry about it.

abhinandan13jan replied:
well, I think it is elaborate enough. `taskRunSnippetMessage` which is fundamental message block for both the taskSnippets. So, if you look close, the final output of both the LogSnippet a PipelineRun/ TaskRun is to return a message as per the relevant taskRun container/pod or a static one.

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/logs/log-snippet-types.ts:
```diff
@@ -0,0 +1,24 @@
+
+export type CombinedErrorDetails = ErrorDetailsWithLogName | ErrorDetailsWithStaticLog;
+
+export type LogSnippetTask = K8sResourceKind & {
```
Nit... My only reasoning for this would be so that it's metadata the reader of the code can gleam and know it's not all that K8s has to offer. 
```suggestion
type LoggablePipelineKinds = PipelineRun | TaskRunKind;
export type LogSnippetTask = LoggablePipelineKinds & {
```

andrewballantyne replied:
Pretty sure this is moot now based on one of my comments, but I want to leave it for clarity and if I am wrong, I still want to report the Nit 😄

abhinandan13jan replied:
It won't work as the earlier implementation of TaskRunKind is all messy

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/logs/__tests__/log-snippet-utils.spec.ts:
```diff
@@ -0,0 +1,53 @@
+import { taskRunSnippetMessage } from '../log-snippet-utils';
+import { ErrorDetailsWithLogName, ErrorDetailsWithStaticLog } from '../log-snippet-types';
+
+const testConatiner = 'container-B';
```
Typo...
```suggestion
const testContainer = 'container-B';
```

abhinandan13jan replied:
updated

andrewballantyne commented on frontend/packages/dev-console/src/components/taskruns/logs/taskRunLogSnippet.ts:
```diff
@@ -0,0 +1,58 @@
+const joinConditions = (conditions: Condition[]) =>
+  conditions.map((condition) => condition.message).join('\n') || 'Unknown failure condition';
+
+export const getLogSnippet = (taskRun: TaskRunKind): TaskRunErrorDetails => {
```
I'm confused... you refactored the PipelineRunLogSnippet to be more able to work with both, why is this still the old (almost)duplicated format?

It appears this is also used by `TaskRunDetailsErrorLog`. My point in my last review was to merge these two, which I thought you were doing with the changes to `pipelineRunLogSnippet.ts`.

What am I missing?

abhinandan13jan replied:
For PLR -> Check the PLR status -> search for the relevant TaskRun which failed -> return correct message based on TaskRun
For TaskRun -> Check TaskRun status -> return correct message based on the TaskRun

The last step in both is the same. Hence, I have written that code in the `log-util` file and rest is separate.

abhinandan13jan replied:
For `PipelineRunDetailsErrorLog` vs `TaskRunDetailsErrorLog`, if you go by the previous comments, it was decided that the two be kept separate. Earlier we were using the same component.

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/logs/pipelineRunLogSnippet.ts:
```diff
@@ -4,26 +4,11 @@ import {
-  conditions.map((condition) => condition.message).join('\n') || 'Unknown failure condition';
-
-export const getLogSnippet = (pipelineRun: PipelineRun): PipelineRunErrorDetails => {
+export const getLogSnippet = (pipelineRun: PipelineRun): CombinedErrorDetails => {
```
Would appear I failed to write tests for this method. Would you mind doing that as part of this PR?

abhinandan13jan replied:
This PR is dragging for a long time. Maybe it'll be better to write under later tech debts, as there are lots of other pending stories

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/logs/log-snippet-utils.ts:
```diff
@@ -0,0 +1,25 @@
+const joinConditions = (conditions: Condition[]) =>
+  conditions.map((condition) => condition.message).join('\n') || 'Unknown failure condition';
+
+export const taskRunSnippetMessage = (
+  taskRun: LogSnippetTask,
+  containerName: string,
+): CombinedErrorDetails => {
+  const taskName = taskRun.pipelineTaskName || taskRun.metadata.name;
```
Hmm, these resources are so alike, but just different enough to dirty the implementation here.

I wonder if instead of passing a `LogSnippetTask` we could look to pass the status object and the name separately, so this internal method is unaware of the relationship between these two separate but same data structures.

Ie:
```suggestion
export const taskRunSnippetMessage = (
  taskName: string,
   // we may want to keep `taskRunStatus` as a stand alone type, but I expanded inline to show what I am referring to
  taskRunStatus: { podName: string, conditions: ConditionType[] },
  containerName: string,
): CombinedErrorDetails => {
  const taskName = taskRun.pipelineTaskName || taskRun.metadata.name;
```

abhinandan13jan replied:
That was the case in the very beginning of the PR. And there would have been no need to keep a separate util for rendering out of  strings.

abhinandan13jan replied:
updated

andrewballantyne reviewed (changes requested):
Closer, but I'm confused a bit on the existence of `taskRunLogSnippet.ts`

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/logs/RunDetailsErrorLog.tsx:
```diff
@@ -1,15 +1,12 @@
 
 type PipelineStatusLogProps = {
-  pipelineRun: PipelineRun;
+  logDetails;
```
Need a type

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/logs/log-snippet-types.ts:
```diff
@@ -0,0 +1,25 @@
+
+export type CombinedErrorDetails = ErrorDetailsWithLogName | ErrorDetailsWithStaticLog;
+
+export type TaskRunStatus = {
+  podName?: string;
+  conditions: Condition[];
+};
+
+export type TaskRunData = K8sResourceKind & {
+  status: TaskRunStatus;
+};
```
Should just add these to the `TaskRunKind`...

andrewballantyne commented on frontend/packages/dev-console/src/components/taskruns/TaskRunDetails.tsx:
```diff
@@ -1,10 +1,17 @@
 
 export interface TaskRunDetailsProps {
-  obj: K8sResourceKind;
+  obj: K8sResourceKind & { status: TaskRunStatus };
```
```suggestion
  obj: TaskRunKind;
```

andrewballantyne commented on frontend/packages/dev-console/src/components/taskruns/logs/taskRunLogSnippet.ts:
```diff
@@ -0,0 +1,30 @@
+import { CombinedErrorDetails, TaskRunData } from '../../pipelineruns/logs/log-snippet-types';
+import { taskRunSnippetMessage } from '../../pipelineruns/logs/log-snippet-utils';
+
+export const getTRLogSnippet = (taskRun: TaskRunData): CombinedErrorDetails => {
```
```suggestion
export const getTRLogSnippet = (taskRun: TaskRunKind): CombinedErrorDetails => {
```

andrewballantyne reviewed (approved):
Looks good, just need to use the `TaskRunKind`

karthikjeeyar commented:
/retest

karthikjeeyar commented:
/retest

karthikjeeyar commented:
/retest

abhinandan13jan commented:
/test backend

karthikjeeyar commented:
Verified locally, works fine
/lgtm

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/logs/log-snippet-types.ts:
```diff
@@ -0,0 +1,28 @@
+
+export type CombinedErrorDetails = ErrorDetailsWithLogName | ErrorDetailsWithStaticLog;
+
+export type TaskRunStatus = {
+  completionTime?: string;
+  conditions: Condition[];
+  podName?: string;
+  startTime?: string;
+  steps?: PLRTaskRunStep[];
+};
+
+export type TaskRunData = K8sResourceKind & {
+  status: TaskRunStatus;
+};
```
Failed to remove this code 😄  Please trim it out so we don't have confusion on which to use.

abhinandan13jan replied:
yup.. updated..thanks

andrewballantyne commented:
/lgtm cancel

andrewballantyne commented:
/approve

Remove the unused code and we should be good to go.

karthikjeeyar commented:
/lgtm
